### ISSUE TITLE
multiple lock icons, and broken? json file? Weird fun...

### ISSUE THREAD

**Issue by ksdavidc** [alberti42/obsidian-plugins-annotations#10]
I was getting 2 “Lock” icons. However, this went away when I locked and unlocked. 

But then it came back with a vengeance. I noticed it when I was making a different change, i.e. trying to create a CSS snippet for the plugin (I hope to add FR elsewhere for this) and also posting about [the loss of editability on settings search](https://github.com/alberti42/obsidian-plugins-annotations/issues/5#issuecomment-2297460505).

It could be related to another behavior I noticed but ignored, which was that several times the "plugin updater" program listed this as a plugin to be updated, and when I tried to update, the update screen disappeared without signifying anything. I restarted and was eventually able to update to 6.5. It could have been that Obsidian hadn't fully loaded at this point (large vault and dataview/omnisearch are slow). Did it get broken earlier?

<img width="795" alt="Screenshot 2024-08-20 at 5 19 16 AM" src="https://github.com/user-attachments/assets/99e8c974-9f7b-4287-94ed-06db56064c8f">

BTW, the right icon seems to work, not the left. 

This sometimes disappears when I click and return. Or appears and then corrects itself. Meanwhile, the json stays the same. Erratic, but for now seems to happen consistently when first switching into settings, as in linked video (first part is switching out of settings, blank, then multiple, then single).
https://github.com/user-attachments/assets/436dcccb-9cc9-4b4f-9de3-d60989c19eb6
Sometimes it is more persistent. 

It is happening in 2 different vaults. In one it goes from 2 icons to 1. In the other, it goes from 1 icon to 2. Sometimes I even get 3! Closing and opening seem to fix it, temporarily...The weirdness never ends.

I went to the JSON file and found what look like duplicate entries…but I hesitate to mess with it.
I do notice that the entries for the current notes (the first one has the correct label_… values for my vaults). 

[data.json](https://github.com/user-attachments/files/16665264/data.json)

```

{
  "annotations": {
    "obsidian-2hop-links-plugin": {
      "name": "2Hop Links Plugin",
      "desc": "This plugin will display the page linked 2hop ahead. It displays the link at the bottom of the Markdown editor. With this plugin, you can roam the digital garden freely.**"
    },
    

etc.

    "unitade": {
      "name": "UNITADE.md",
      "desc": "used for smart pdfs from better pdf +++ plugin**"
    }
  },
  "plugins_annotations_uuid": "BC56AB7B-A46F-4ACF-9BA1-3A4461F74C79",
  "hide_placeholders": true,
  "delete_placeholder_string_on_insertion": true,
  "label_mobile": "<b>Annotation:&nbsp;</b>",
  "label_desktop": "[[admin/Plugins/${plugin_name}| ${plugin_name}]]: **",
  "label_placeholder": "*Click to Make Annotation* for [[admin/Plugins/${plugin_name}| ${plugin_name}]]",
  "editable": true,
  "automatic_remove": false,
  "markdown_file_path": "Admin/PluginAnnotations.md",
  "backups": [
    {
      "name": "Settings before upgrade from 1.5 to 1.6",
      "date": "2024-08-18T08:13:53.886Z",
      "settings": {
        "annotations": {
          "obsidian-admonition": {
            "desc": "test",
            "name": "Admonition",
            "type": "markdown"
          },
          "advanced-paste": {
            "desc": "this is another",
            "name": "Advanced Paste",
            "type": "markdown"
          }
        },
        "plugins_annotations_uuid": "BC56AB7B-A46F-4ACF-9BA1-3A4461F74C79",
        "hide_placeholders": false,
        "delete_placeholder_string_on_insertion": true,
        "label_mobile": "<b>Annotation:&nbsp;</b>",
        "label_desktop": "**Note**: ",
        "label_placeholder": "Click to *add note*:",
        "editable": true,
        "automatic_remove": false,
        "markdown_file_path": "Admin/PluginAnnotations.md",
        "compatibility": "1.5.0"
      }
    },
    {
      "name": "Settings before upgrade from 1.4 to 1.5",
      "date": "2024-08-15T04:14:12.256Z",
      "settings": {
        "annotations": {
          "obsidian-admonition": {
            "anno": "test",
            "name": "Admonition"
          }
        },
        "plugins_annotations_uuid": "B265C5B2-A6AD-4194-9E4C-C1327DB1EA18",
        "hide_placeholders": false,
        "delete_placeholder_string_on_insertion": false,
        "label_mobile": "<b>Annotation:&nbsp;</b>",
        "label_desktop": "<b>Personal annotation:&nbsp;</b>",
        "label_placeholder": "<em>Add your personal comment about <strong>${plugin_name}</strong> here...</em>",
        "editable": true,
        "automatic_remove": false
      }
    }
  ],
  "compatibility": "1.6.0"
}

```

**Comment by alberti42** [comment]
Thanks a lot for the detailed feedback. Uff. I was hoping we were done and reached a mature state but apparently there are many things still to be fixed.

> listed this as a plugin to be updated, and when I tried to update, the update screen disappeared without signifying anything. 

I am suspecting that it shows the plugin as updatable when I update the [manifest.json](https://github.com/alberti42/obsidian-plugins-annotations/blob/main/manifest.json) file with a new version. However, if I do not publish a new release [here](https://github.com/alberti42/obsidian-plugins-annotations/releases/) with the same version, it is for sure not possible to have a full successful installation. If my supposition is correct, this would explain why you could not complete the update in the past... because often I updated the `manifest.jon` file days before properly releasing a new version. If it is so, it is weird that Obsidian looks in the `manifest.json` instead of looking at the proper list of the releases.

I assume that you were now able to install the most recent release, version 1.6.3. If so, then I would say the problem with the installation is resolved and I need to look into the other problems.

**Comment by ksdavidc** [comment]
yes installed.

**Comment by alberti42** [comment]
> I went to the JSON file and found what look like duplicate entries…but I hesitate to mess with it.

I did not see any duplicate entry in the example you copied and pasted. Could you be specific? Perhaps I missed the problem.

If you refer to the entries associated with the backups, well, it is normal that they are duplicated annotations for the same plugin since this is a backup. Backups are just mere copy of the settings.

**Comment by alberti42** [comment]
> It is happening in 2 different vaults. In one it goes from 2 icons to 1. In the other, it goes from 1 icon to 2. Sometimes I even get 3! Closing and opening seem to fix it, temporarily...The weirdness never ends.

I can see that it can be frustrating. It is probably related to a conflict with the other plugin you are using that also shows extra icons there in the preference pane. What are these other plugins that I could test?

As a temporary fix, I could add a toggle in the preference pane that shows/hides icons in the community plugins preference pane. You can still change the editability of your notes, but only from the settings of my plugin. Not just at a distance of a click but it could be a better solution until we find the real problem.

**Comment by ksdavidc** [comment]
> > I went to the JSON file and found what look like duplicate entries…but I hesitate to mess with it.
> 
> I did not see any duplicate entry in the example you copied and pasted. Could you be specific? Perhaps I missed the problem.
> 
> If you refer to the entries associated with the backups, well, it is normal that they are duplicated annotations for the same plugin since this is a backup. Backups are just mere copy of the settings.

What I meant by duplicate entries was not duplicate annotations, but this part at the end that are not quite "duplicate" but very similar, both being "plugins_annotations_uuid:" entries, though at different levels in the json. Perhaps they are supposed ot be that way??



```
  "plugins_annotations_uuid": "BC56AB7B-A46F-4ACF-9BA1-3A4461F74C79",
        "hide_placeholders": false,
        "delete_placeholder_string_on_insertion": true,
        "label_mobile": "<b>Annotation:&nbsp;</b>",
        "label_desktop": "**Note**: ",
        "label_placeholder": "Click to *add note*:",
        "editable": true,
        "automatic_remove": false,
        "markdown_file_path": "Admin/PluginAnnotations.md",
        "compatibility": "1.5.0"
      }
    },
    {
      "name": "Settings before upgrade from 1.4 to 1.5",
      "date": "2024-08-15T04:14:12.256Z",
      "settings": {
        "annotations": {
          "obsidian-admonition": {
            "anno": "test",
            "name": "Admonition"
          }
        },
        "plugins_annotations_uuid": "B265C5B2-A6AD-4194-9E4C-C1327DB1EA18",
        "hide_placeholders": false,
        "delete_placeholder_string_on_insertion": false,
        "label_mobile": "<b>Annotation:&nbsp;</b>",
        "label_desktop": "<b>Personal annotation:&nbsp;</b>",
        "label_placeholder": "<em>Add your personal comment about <strong>${plugin_name}</strong> here...</em>",
        "editable": true,
        "automatic_remove": false
      }
    }
  ],
  "compatibility": "1.6.0"
```

**Comment by alberti42** [comment]
This appears to be normal. However, probably the settings you provided in the last post was edited by you in the copy and paste where you remove some parts. 

The full [json](https://github.com/user-attachments/files/16665264/data.json) file you posted before is correct. If you notice, there is an entry at the main level called `"backups": [`. In square brackets is a list of backups containing a name, time stamp, and the copy of the settings.

So I believe the json file is not broken but as expected.

The question of the multiple icons is still to be investigated.

**Comment by alberti42** [comment]
I found another small bug, which I corrected in the version 1.6.4. This gave me the possibility to test my supposition:

> I am suspecting that it shows the plugin as updatable when I update the [manifest.json](https://github.com/alberti42/obsidian-plugins-annotations/blob/main/manifest.json) file with a new version. However, if I do not publish a new release [here](https://github.com/alberti42/obsidian-plugins-annotations/releases/) with the same version, it is for sure not possible to have a full successful installation. If my supposition is correct, this would explain why you could not complete the update in the past... because often I updated the manifest.jon file days before properly releasing a new version. If it is so, it is weird that Obsidian looks in the manifest.json instead of looking at the proper list of the releases.

I was indeed correct. Issuing a new version in the `manifest.json` induces Obsidian headquarter to believe that there is a new version ready. But this is not the case until a release is properly published in github. I take this as a bug in the obsidian workflow. Anyhow, now that I know about it, it is easy to avoid the problem in the future, but in my view, Obsidian's decision to look for the manifest.json instead of the list of releases makes no sense.

**Comment by ksdavidc** [comment]
> > It is happening in 2 different vaults. In one it goes from 2 icons to 1. In the other, it goes from 1 icon to 2. Sometimes I even get 3! Closing and opening seem to fix it, temporarily...The weirdness never ends.
> 
> I can see that it can be frustrating. It is probably related to a conflict with the other plugin you are using that also shows extra icons there in the preference pane. What are these other plugins that I could test?
> 
> As a temporary fix, I could add a toggle in the preference pane that shows/hides icons in the community plugins preference pane. After you hid the lock icons, you can still change the editability of your notes, but only from the settings of my plugin. Not just at a distance of a click but it could be a better solution until we find the real problem.

Not that frustating. As far as the main functionality, everything is fine. I think the toggle is unnecessary. As I said, things work, but in an odd way is all. I just wanted you to know that there may be something awry. Maybe not, maybe this is how it works, period. That's okay, too.

The other plugin that is showing icons is "divide and conquer". 


YES! Turning it off does seem, at first blush, to fix the problem.

**Comment by ksdavidc** [comment]
> This appears to be normal. However, probably the settings you provided in the last post was edited by you in the copy and paste where you remove some parts.
> 
> The full [json](https://github.com/user-attachments/files/16665264/data.json) file you posted before is correct. If you notice, there is an entry at the main level called `"backups": [`. In square brackets is a list of backups containing a name, time stamp, and the copy of the settings.
> 
> So I believe the json file is not broken but as expected.
> 
> The question of the multiple icons is still to be investigated.

Yes, I had edited it for length, and I am reassured that the actual file is okay...

**Comment by ksdavidc** [comment]
> I found another small bug, which I corrected in the version 1.6.4. This gave me the possibility to test my supposition:
> 
> > I am suspecting that it shows the plugin as updatable when I update the [manifest.json](https://github.com/alberti42/obsidian-plugins-annotations/blob/main/manifest.json) file with a new version. However, if I do not publish a new release [here](https://github.com/alberti42/obsidian-plugins-annotations/releases/) with the same version, it is for sure not possible to have a full successful installation. If my supposition is correct, this would explain why you could not complete the update in the past... because often I updated the manifest.jon file days before properly releasing a new version. If it is so, it is weird that Obsidian looks in the manifest.json instead of looking at the proper list of the releases.
> 
> I was indeed correct. Issuing a new version in the `manifest.json` induces Obsidian headquarter to believe that there is a new version ready. But this is not the case until a release is properly published in github. I take this as a bug in the obsidian workflow. Anyhow, now that I know about it, it is easy to avoid the problem in the future, but in my view, Obsidian's decision to look for the manifest.json instead of the list of releases makes no sense.
> 
> I believe this fully explains why you encountered problems in the past to install the plugin.

It wasn't really a dealbreaker, but it is nice to have things working cleanly. I see your point, but who knows, maybe obsidian has a reason (speed, etc.) for doing it the way they do. I'll see if things work when the update comes.

**Comment by alberti42** [comment]
Thanks a lot for the feedback. I released a version 1.6.5 with a fix for the multiple icons. I found a small bug that became quite visible when using `Conquer & Divide` plugin:

> The new release should fix the bug of multiple lock icons being created when the plugin `Divide & Conquer` was used. The plugin disables and re-enables plugins. Every time this plugin was reloaded, a new icon was created because the previous one was never removed.
